src/noding/MCIndexSegmentSetMutualIntersector.cpp

```diff
@@ -58,15 +61,15 @@ MCIndexSegmentSetMutualIntersector::intersectChains()
     MCIndexSegmentSetMutualIntersector::SegmentOverlapAction overlapAction(*segInt);
 
     std::vector<void*> overlapChains;
-    for(const auto& queryChain : monoChains) {
+    for(auto& queryChain : monoChains) {
```

Comment:
Do  you need queryChain to be non-const ?

Reply:
Maybe? Not sure how deep into the guts I would have to follow the const chain.

Reply:
It starts running deep into NodedSegmentString, and NSS definitely is not const all the time. Maybe the whole thing should be non-const. https://github.com/pramsey/geos/pull/1

Reply:
I only asked becuase it _was_ const before, apparently, thus I was wondering why you had to turn into non-const here.